FreeRTOS/FreeRTOS-Kernel
# Add Trace Hook Macros to all API calls

Add Trace Hook Macros to all API calls

Description
-----------
This pull-request adds out-of-the-box support for different tracing tools.
For more information see following forum post: https://forums.freertos.org/t/add-tracing-functionality-for-all-api-calls/18007.

New trace hook macros have been added for all public API functions, one for each function entry and one for each exit.
There are no functional changes, as the macros are by default empty.

Test Steps
-----------
N/A

Checklist:
----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
